dowhy/gcm/stats.py

```diff
@@ -161,3 +164,65 @@ def permute_features(feature_samples: np.ndarray,
             np.random.shuffle(feature_samples[:, feature])
 
     return feature_samples
+
+
+def estimate_f_test_p_value(X_training_a: np.ndarray,
+                            X_training_b: np.ndarray,
+                            Y_training: np.ndarray,
+                            X_test_a: np.ndarray,
+                            X_test_b: np.ndarray,
+                            Y_test: np.ndarray) -> float:
+    """ Estimates the p-value for the nullhypothesis that the same regression curve with less parameters can be
+    achieved. This is, a linear model trained on a data set A with d number of features has the same performance
+    (in terms of squared error) relative to the number of features as a model trained on a data set B with k number
+    features, where k < d. Here, both data sets need to have the same target values. A small p-value would
+    indicate that the model performances are significantly different.
+
+    Note that all given test samples are utilized in the f-fest.
+
+    See https://en.wikipedia.org/wiki/F-test#Regression_problems for more details.
+
+    :param X_training_a: Input training samples for model A.
+    :param X_training_b: Input training samples for model B.
+    :param Y_training: Target training values.
+    :param X_test_a: Test samples for model A.
+    :param X_test_b: Test samples for model B.
+    :param Y_test: Test values.
+    :return: A p-value on [0, 1].
+    """
+    X_training_a, X_test_a = shape_into_2d(X_training_a, X_test_a)
+
+    if X_training_b.size > 0:
+        X_training_b, X_test_b = shape_into_2d(X_training_b, X_test_b)
+    else:
+        X_training_b = X_training_b.reshape(0, 0)
+        X_test_b = X_test_b.reshape(0, 0)
+
+    if X_training_a.shape[1] <= X_training_b.shape[1]:
+        raise ValueError("Data for A should have more dimensions (i.e. model parameters) than the data for B!")
```

Comment:
Here too, the error message can be more informative. "E.g., Data for baseline features (B) should have more dimensions (model parameters) than the data for the all features."

Reply:
The data with all features should have more dimensions than the baseline data. I think the term "baseline" might be a bit confusing here. Strictly speaking, training data A and training data B could come from completely different subsets of features, as long as A has more.